src/polymer/expression-scanner.ts

```diff
@@ -128,6 +138,67 @@ export class DatabindingExpression {
     return correctSourceRange(
         databindingRelativeSourceRange, this.locationOffset);
   }
+
+  private _extract() {
+    if (this._expressionAst.body.length !== 1) {
+      this.warnings.push(this._validationWarning(
+          'Expected exactly one expression', this._expressionAst));
+      return;
+    }
+    const expressionStatement = this._expressionAst.body[0]!;
+    if (expressionStatement.type !== 'ExpressionStatement') {
+      this.warnings.push(this._validationWarning(
+          'ExpressionStatement expected', expressionStatement));
+      return;
+    }
+    let expression = expressionStatement.expression;
+    if (expression.type === 'UnaryExpression') {
+      if (expression.operator !== '!') {
+        this.warnings.push(this._validationWarning(
+            'Only the not (!) operator is supported.', expression));
+        return;
+      }
+      expression = expression.argument;
+    }
+    this._handleSubExpression(expression);
+  }
+
+  private _handleSubExpression(expression: estree.Node) {
```

Comment:
Fair, named more consistently. Also added a flag to the method to track whether call expressions are allowed.